source/server/getting-started/configuration/sponge-conf.rst

```diff
@@ -13,476 +13,1678 @@ Below is a list of available config settings and their comments inside the globa
+  | asynchronously, Sponge will automatically capture said command
+  | and submit it for processing on the server thread. The catch to
+  | this is that some mods are performing these commands in vanilla
+  | without considering the possible consequences of such commands
+  | affecting any thread-unsafe parts of Minecraft, such as worlds,
+  | block edits, entity spawns, etc. If this is set to false, Sponge
+  | will politely ignore the command being executed, and emit a warning
+  | about said command anyways.
+  | **Type:** ``boolean``
+  | **Default:** ``true``
+  |
+
+* **verbose**
+
+  | If ``true``, the phase tracker will print out when there are too many phases 
+  | being entered, usually considered as an issue of phase re-entrance and 
+  | indicates an unexpected issue of tracking phases not to complete. 
+  | If this is not reported yet, please report to Sponge. If it has been 
+  | reported, you may disable this.
+  | **Type:** ``boolean``
+  | **Default:** ``true``
+  |
+
+* **verbose-errors**
+
+  | If ``true``, the phase tracker will dump extra information about the current phases 
+  | when certain non-PhaseTracker related exceptions occur. This is usually not necessary, as the information 
+  | in the exception itself can normally be used to determine the cause of the issue
+  | **Type:** ``boolean``
+  | **Default:** ``false``
+  |
+
+.. _ConfigType_Commands:
+
+commands (Commands)
+-------------------
+
+* **aliases**
+
+  | Command aliases will resolve conflicts when multiple plugins request a specific command, 
+  | Correct syntax is <unqualified command>=<plugin name> e.g. ``sethome=homeplugin``
+  | **Type:** ``Map<String, String>``
+  |
+
+* **command-hiding**
+
+  | Defines how Sponge should act when a user tries to access a command they do not have
+  | permission for
+  | **Type:** :ref:`CommandsHidden<ConfigType_CommandsHidden>`
+  |
+
+* **enforce-permission-checks-on-non-sponge-commands**
+
+  | Some mods may not trigger a permission check when running their command. Setting this to
+  | true will enforce a check of the Sponge provided permission (``<modid>.command.<commandname>``).
+  | Note that setting this to true may cause some commands that are generally accessible to all to
+  | require a permission to run.
+  | Setting this to true will enable greater control over whether a command will appear in
+  | tab completion and Sponge's help command.
+  | If you are not using a permissions plugin, it is highly recommended that this is set to false
+  | (as it is by default).
+  | **Type:** ``boolean``
+  | **Default:** ``false``
+  |
+
+* **multi-world-patches**
+
+  | Patches the specified commands to respect the world of the sender instead of applying the 
+  | changes on the all worlds.
+  | **Type:** ``Map<String, Boolean>``
+  |
+
+.. _ConfigType_CommandsHidden:
+
+commands.command-hiding (CommandsHidden)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
I'm not sure if Sphinx will care, but you are using mixed title types a bit (`-----`, `^^^^^`, `""""""`).
The form `------` is the most common, so maybe stick with that for second-tier headers.

Reply:
This is a byproduct of the suggestion to use the path as the title. It uses different headline levels, but the difference is mostly invisible.

````html
<h3>GlobalConfig</h3>

<h4>commands (Commands)</h4>

<h5>commands.command-hiding (CommandsHidden)</h5>
````
Should I change it to use the same level?

Reply:
I didn't see any significant differences in the preview, hence my query if Sphinx even cares. If it comes up anywhere, it will be indexing. If you think it's OK, then we're about ready to pull the lever.

Reply:
I would like to keep the different header levels, so we could customize them if needed.
Currently it's hard to see but the config is a single tree, where every type is only used once. But if the config changes in a way that re-uses a datatype, then that type will automatically be removed from the main tree and moved to a separate tree. The separate tree will then of course start from h3 again (breaking the current impression of an alphabetical order).